### PR TITLE
INI: Swap out `header` for `section`

### PR DESCRIPTION
**Pull request by hoonweiting**
This PR replaces the `header` token with `section`, bringing INI in line with editorconfig and systemd.

### CODE REVIEW

File: tests/languages/ini/section_feature.test [hunk 1]
```diff
@@ -1,4 +1,4 @@
- [foo1]
```
Comment (RunDevelopment) [inline tests/languages/ini/section_feature.test:1]: Please revert this line. This tests that sections surrounded by spaces are still highlighted correctly. This is part of INI and used to be a bug IIRC.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

**Comment by RunDevelopment** [comment]
Thank you for all the great PRs, and sorry for the delay. 

Just one minor nit with this one :)

**Comment by hoonweiting** [comment]
It's great to see you! Don't worry about any delay, hope you've been doing well :D

**Review by RunDevelopment — APPROVED** [review]

**Comment by RunDevelopment** [comment]
Thank you @hoonweiting!

**Pull request merged**